### PR TITLE
[Tests] Improve help and failure mode

### PR DESCRIPTION
**Pull request by newling**
Before if you passed an invalid string for target_device, the tests passed without running anything

### CODE REVIEW

File: build_tools/ci/cpu_comparison/run.py [hunk 1]
```diff
@@ -1123,9 +1123,34 @@ def all_tests(
...
+        "xcvc1902",
+        "xcve2302",
+        "xcve2802",
+        "npu1",
+        "npu1_1col",
+        "npu1_2col",
+        "npu1_3col",
+        "npu1_4col",
+        "npu4",
+    ]
```
Comment (Abhishek-Varma) [inline build_tools/ci/cpu_comparison/run.py:1148]: I'm not sure if we've tested for all these devices.

Would it be okay to just go with `npu1_4col` and `npu4` as the two accepted devices ?

For instance, `run_on_target` has been defaulted to be `npu1_4col` for all tests and just one test adds `npu4` to the same list.

So, there should just be a correspondence between the two, I believe.

**Review by jtuyls — APPROVED** [review]
LGTM

File: build_tools/ci/cpu_comparison/run.py (hunk 1 above)
Comment (Abhishek-Varma) [reply build_tools/ci/cpu_comparison/run.py:1148]: Okay okay, that makes sense.

**Review by Abhishek-Varma — APPROVED** [review]
LGTM!

**Pull request merged**